### PR TITLE
1060 assam cdl

### LINKED ISSUE #1060: Add CDL Assam spider
While CDL presents OCDS data at https://assam.open-contracting.in/datasets, actual datasets (OCDS package & flattened data) are stored on GitHub

**Links to bulk download and/or API documentation**
https://github.com/CivicDataLab/assam-tenders-data/tree/main/data

**Links to bulk download and/or API endpoints**
https://github.com/CivicDataLab/assam-tenders-data/blob/main/data/ProcessedData/ocds-mapped-data/current/ocds_mapped_data.json.zip

**Priority**
anytime

**Data structure**
Data has been checked using DRT (and DRT was able to process it)

**Publication format**
Data is provided as an archive that contains package.json file with compiled releases packaged.

### PR DESCRIPTION
**Pull request by yolile**
closes #1060

### CODE REVIEW

File: kingfisher_scrapy/base_spiders/compressed_file_spider.py [hunk 1]
```diff
@@ -20,6 +20,10 @@ class CompressedFileSpider(BaseSpider):
     #. Optionally, add a ``resize_package = True`` class attribute to split large packages (e.g. greater than 100MB)
     #. Optionally, add a ``yield_non_archive_file = True`` class attribute if the spider requests both archive files
        and JSON files. Otherwise, the spider raises an ``UnknownArchiveFormatError`` exception.
+    #. Optionally, add a ``file_name_must_contain = 'text'`` class attribute to only decompress the files which names
+       contain the given text.
+    #. Optionally, add a ``file_name_must_not_contain = 'text'`` class attribute to only decompress the files which
+       name do not contain the given text.
```
Comment (jpmckinney) [inline kingfisher_scrapy/base_spiders/compressed_file_spider.py:26]:
```suggestion
    #. Optionally, add a ``file_name_must_contain = 'text'`` class attribute to only decompress the files whose names
       contain the given text.
    #. Optionally, add a ``file_name_must_not_contain = 'text'`` class attribute to only decompress the files whose
       names do not contain the given text.
```

File: kingfisher_scrapy/base_spiders/compressed_file_spider.py [hunk 2]
```diff
@@ -75,7 +80,8 @@ def parse(self, response):
 
             filename = file_info.filename
             basename = os.path.basename(filename)
-            if self.file_name_must_contain not in basename:
+            if self.file_name_must_contain not in basename or (self.file_name_must_not_contain and
+                                                               self.file_name_must_not_contain in basename):
```
Comment (jpmckinney) [inline kingfisher_scrapy/base_spiders/compressed_file_spider.py:84]:
```suggestion
            if (
                self.file_name_must_contain not in basename
                or (self.file_name_must_not_contain and self.file_name_must_not_contain in basename)
            ):
```

File: kingfisher_scrapy/spiders/india_assam_civic_data_lab.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    Domain
+      India Assam - CivicDataLab
+    Bulk download documentation
+      https://github.com/CivicDataLab/assam-tenders-data/tree/main/data
+    """
+    name = 'india_assam_civic_data_lab'
+
+    # CompressedFileSpider
+    # The archived file contains a __MACOSX directory with a temporary ._package.json file.
+    file_name_must_not_contain = '._'
```
Comment (jpmckinney) [inline kingfisher_scrapy/spiders/india_assam_civic_data_lab.py:17]: `__MACOSX` is a common occurrence (old macOS versions always added it, when compressing files using the Finder). So, I prefer to exclude files whose `filename.startswith('__MACOSX')` in CompressedFileSpider.

That said, we can keep the new "not contains" logic in CompressedFileSpider

Comment (yolile) [reply kingfisher_scrapy/spiders/india_assam_civic_data_lab.py:17]: Fair enough and done

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**